src/components/CourseInfoRow.js

```diff
@@ -711,37 +665,50 @@ function CourseInfoRow({ courseInfo, selected, isfavorite, displayTable }) {
               handleAddFavorite(courseInfo.id);
               reportEvent(
                 "course_info_row",
-                isfavorite ? "remove_favorite" : "add_favorite",
+                isFavorite ? "remove_favorite" : "add_favorite",
                 courseInfo.id
               );
             }}
             isLoading={addingFavoriteCourse}
           >
             <Box>
-              {<Icon as={isfavorite ? FaHeart : FaRegHeart} boxSize="4" />}
+              {<Icon as={isFavorite ? FaHeart : FaRegHeart} boxSize="4" />}
             </Box>
           </Button>
-          <Button
-            size="sm"
-            ml={{ base: 0, md: "10px" }}
-            colorScheme={selected ? "red" : "blue"}
-            onClick={() => {
-              addCourseToTable(courseInfo);
-              reportEvent(
-                "course_info_row",
-                selected ? "remove_course" : "add_course",
-                courseInfo.id
-              );
-            }}
-            isLoading={addingCourse}
+          <Tooltip
+            label="非當學期課程"
+            hasArrow
+            shouldWrapChildren
+            placement="top"
+            isDisabled={
+              courseInfo.semester === process.env.NEXT_PUBLIC_SEMESTER
+            }
           >
-            <Box
-              transform={selected ? "rotate(45deg)" : ""}
-              transition="all ease-in-out 200ms"
+            <Button
+              size="sm"
+              ml={{ base: 0, md: "10px" }}
+              colorScheme={selected ? "red" : "blue"}
+              onClick={() => {
+                addCourseToTable(courseInfo);
+                reportEvent(
+                  "course_info_row",
+                  selected ? "remove_course" : "add_course",
+                  courseInfo.id
+                );
+              }}
+              isLoading={addingCourse}
+              disabled={
+                courseInfo.semester !== process.env.NEXT_PUBLIC_SEMESTER
+              }
             >
-              <FaPlus />
-            </Box>
-          </Button>
+              <Box
+                transform={selected ? "rotate(45deg)" : ""}
+                transition="all ease-in-out 200ms"
```

Comment:
I've added a button control and a tooltip here to prevent adding non-active courses to active course tables.